KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift

```diff
@@ -10,110 +10,109 @@ import Foundation
 import KarhooSDK
 import UIKit
 import SwiftUI
+import Combine
 
 enum NewCheckoutState {
     case loading
-    case asap
+    case idle
```

Comment:
I don't like this state name. It's not clear that it's an alternative to scheduled. To me it sounds that nobody is interacting with the screen and it's not locked by design (aka `loading`). The checkout can be both idle and scheduled or idle and asap. If you have a reason for this name change, please document it.

Reply:
I was thinking the State should not be aware of asap/schedules, since it does not affect the presentation logic anyhow. That's why I introduced `idle` (or any other `the screen is ready for user interaction` name).

Reply:
I agree, but that means removing `scheduled` from the list of cases. If checkout user interactivity is the scope of this enum, we can have `loading`, `idle`, and `gatheringInfo` (for when the passenger details of flight number or another modal is open). The name for the last case is debatable as long as it's descriptive of what it's actually used for. Thoughts?

Reply:
enum NewCheckoutState {
    /// Screen is locked and user's interactions are disabled. Used for crucial data loading (like payment confirmation)
    case loading
    /// UI is enabled but some data's missing in order to proceed with booking, so user needs to provide them.
    case gatheringInfo
    /// UI is enabled and all data's in place. Scene is waiting for user's confirmation.
    case readyToBook
}